src/core/Table/EditableCell.tsx

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { Cell, TableCellProps } from 'react-table';
+
+export type OnCellEditCallbackParams<T extends Record<string, unknown>> = {
+  columnId: string;
+  value: string;
+  rowData: T;
+};
+
+export type EditableCellProps<T extends Record<string, unknown>> = {
+  cell: Cell<T>;
+  cellProps: TableCellProps;
+  onCellEdit?: (cellEditParams: OnCellEditCallbackParams<T>) => void;
+  children: React.ReactNode;
+};
+
+const EditableCell = <T extends Record<string, unknown>>(
+  props: EditableCellProps<T>,
+) => {
+  const { cell, cellProps, onCellEdit, children } = props;
+
+  const [value, setValue] = React.useState(cell.value);
+  const [isDirty, setIsDirty] = React.useState(false);
```

Comment:
Is this a standard name? Something like `didChange` or `isChanged` or `isModified` might be more clear.

Or add a comment.

Reply:
I know Angular forms had this field `isDirty`